### ISSUE TITLE
Fan in and fan out can not be computed for tensor with less than 2 dimensions??

### ISSUE THREAD

**Issue by VisionZQ** [ShuangXieIrene/ssds.pytorch#9]
Hi,

   I train the mode of 'fssd_lite_mobilenetv2_train_voc.yml',  ValueError 'Fan in and fan out can not be computed for tensor with less than 2 dimensions'
<img width="773" alt="wx20180408-123216 2x" src="https://user-images.githubusercontent.com/22951577/38463254-e9e2d966-3b28-11e8-941d-96daa2b25b3a.png">

    I try to review code ,but I can not handle the issue by change the method of init.  Can you give me some proposal？thanks

**Comment by ShuangXieIrene** [comment]
Sorry for the late reply, Recently, we are little busy with another project. It's almost done, so we'll have time to come back and play with different detector again.

I do not analyze that part of the code in fact. But we fixed it by applying just simply remove the init() from the original code by the most recent commit.

We'll fix it at the later time. Cause better init way may have an impact on the final result. But it is not fundamentally. So you can just use the current version first.

Best,

**Comment by VisionZQ** [comment]
I really appreciate you . I quite agree with you,init the net from the pre-training model is ok。。
I have a little question，
<img width="639" alt="wx20180409-092729 2x" src="https://user-images.githubusercontent.com/22951577/38475135-5725b0ca-3bd8-11e8-8c1b-d86afe0978c8.png">

why did not you normalize the bounding box with width and height ? Are there any tricks here?

**Comment by VisionZQ** [comment]
Hi，
    I review the code again carefully. Seems to have found the bug.
    Initialize the the 'conv' layer, but 'conv' layer involve ['Conv2d , BatchNorm.Relu],
   
<img width="639" alt="wx20180409-092729 2x" src="https://user-images.githubusercontent.com/22951577/39079957-6e474b28-4557-11e8-84aa-e976265dcf68.png">

   The dimension of BatchNorm layer only is 1 , it is less than 2..
   so , 'Fan in and fan out can not be computed for tensor with less than 2 dimensions' happened when initialize the layer by KaiMing method

**Comment by foreverYoungGitHub** [comment]
Brilliant! it should be the problem!

I'll check later, recently the code need to be changed a lot caused the pytorch version update.

Thanks for your great work! I'll check it again.

Best,

**Comment by chuanfanyoudong** [comment]
> Hi，
> I review the code again carefully. Seems to have found the bug.
> Initialize the the 'conv' layer, but 'conv' layer involve ['Conv2d , BatchNorm.Relu],
> 
> <img alt="qq20180421-113029 2x" width="425" src="https://user-images.githubusercontent.com/22951577/39079981-0be50eb0-4558-11e8-8494-f81300b3c1d9.png">
> 
> The dimension of BatchNorm layer only is 1 , it is less than 2..
> so , 'Fan in and fan out can not be computed for tensor with less than 2 dimensions' happened when initialize the layer by KaiMing method

hi，i meet same question with u , i find the problem accure when i want to construct the vectors. could u tell me how to solve this problem？ thank u

**Comment by chuanfanyoudong** [comment]
> > Hi，
> > I review the code again carefully. Seems to have found the bug.
> > Initialize the the 'conv' layer, but 'conv' layer involve ['Conv2d , BatchNorm.Relu],
> > <img alt="qq20180421-113029 2x" width="425" src="https://user-images.githubusercontent.com/22951577/39079981-0be50eb0-4558-11e8-8494-f81300b3c1d9.png">
> > The dimension of BatchNorm layer only is 1 , it is less than 2..
> > so , 'Fan in and fan out can not be computed for tensor with less than 2 dimensions' happened when initialize the layer by KaiMing method
> 
> hi，i meet same question with u , i find the problem accure when i want to construct the vectors. could u tell me how to solve this problem？ thank u,

sorry to disturb u , i change the version of the torchtext , and the problem was solved, the reason is that when the version of torchtext is 0.3.1 , it can't init the vector of "unk", and i change the version to 0.2.3, the problem were solved. so sorry

**Comment by atnlp** [comment]
> sorry to disturb u , i change the version of the torchtext , and the problem was solved, the reason is that when the version of torchtext is 0.3.1 , it can't init the vector of "unk", and i change the version to 0.2.3, the problem were solved. so sorry

Solved my problem

> > > Hi，
> > > I review the code again carefully. Seems to have found the bug.
> > > Initialize the the 'conv' layer, but 'conv' layer involve ['Conv2d , BatchNorm.Relu],
> > > <img alt="qq20180421-113029 2x" width="425" src="https://user-images.githubusercontent.com/22951577/39079981-0be50eb0-4558-11e8-8494-f81300b3c1d9.png">
> > > The dimension of BatchNorm layer only is 1 , it is less than 2..
> > > so , 'Fan in and fan out can not be computed for tensor with less than 2 dimensions' happened when initialize the layer by KaiMing method
> > 
> > 
> > hi，i meet same question with u , i find the problem accure when i want to construct the vectors. could u tell me how to solve this problem？ thank u,
> 
> sorry to disturb u , i change the version of the torchtext , and the problem was solved, the reason is that when the version of torchtext is 0.3.1 , it can't init the vector of "unk", and i change the version to 0.2.3, the problem were solved. so sorry

Solved my problem
